server/api.go

```diff
@@ -462,6 +476,13 @@ func (a *API) oauthRedirectHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	err = a.p.store.DeleteUserInvite(mmUserID)
+	if err != nil {
+		a.p.API.LogWarn("Unable to clear user invite", "error", err.Error())
+		http.Error(w, "Something went wrong", http.StatusInternalServerError)
+		return
```

Comment:
To confirm, do we definitely want to interrupt the rest of the flow should this fail? (And unrelated to the PR, but we have waaay too much logic in the API layer :p.)

Reply:
That's a fair point—if it's gotten this far and only errors here, it would probably be fine to continue on. Will change.

Reply:
Probably beyond scope, but shouldn't this portion of the function be transactional...completely succeed or fail. I didn't completely study it, but with all the error checks and database calls...both storing and retrieving a database error may leave the user in an unknown state.